src/runners/layout.ts

```diff
@@ -0,0 +1,146 @@
+  builtins: string[];
+  rcUnits: number;
+  publicMemoryFraction: number;
+  dilutedPool?: DilutedPool;
+};
+
+export const DEFAULT_DILUTED_POOL: DilutedPool = {
+  unitsPerStep: 16,
+  spacing: 4,
+  nBits: 16,
+};
+
+export const layouts: { [key: string]: Layout } = {
+  plain: {
+    builtins: [],
+    rcUnits: 16,
+    publicMemoryFraction: 4,
+  },
+  small: {
+    builtins: ['output', 'pedersen', 'range_check', 'ecdsa'],
+    rcUnits: 16,
+    publicMemoryFraction: 4,
+  },
+  dex: {
+    builtins: ['output', 'pedersen', 'range_check', 'ecdsa'],
+    rcUnits: 4,
+    publicMemoryFraction: 4,
+  },
+  recursive: {
+    builtins: ['output', 'pedersen', 'range_check', 'bitwise'],
+    rcUnits: 4,
+    publicMemoryFraction: 8,
+    dilutedPool: DEFAULT_DILUTED_POOL,
+  },
+  starknet: {
+    builtins: [
+      'output',
+      'pedersen',
+      'range_check',
+      'ecdsa',
+      'bitwise',
+      'ec_op',
+      'poseidon',
+    ],
+    rcUnits: 4,
+    publicMemoryFraction: 8,
+    dilutedPool: {
+      unitsPerStep: 2,
+      spacing: 4,
+      nBits: 16,
+    },
+  },
+  starknet_with_keccak: {
+    builtins: [
+      'output',
+      'pedersen',
+      'range_check',
+      'ecdsa',
+      'bitwise',
+      'ec_op',
+      'keccak',
+      'poseidon',
+    ],
+    rcUnits: 4,
+    publicMemoryFraction: 8,
+    dilutedPool: DEFAULT_DILUTED_POOL,
+  },
+  recursive_large_output: {
+    builtins: ['output', 'pedersen', 'range_check', 'bitwise', 'poseidon'],
+    rcUnits: 4,
+    publicMemoryFraction: 8,
+    dilutedPool: DEFAULT_DILUTED_POOL,
+  },
+  recursive_with_poseidon: {
+    builtins: ['output', 'pedersen', 'range_check', 'bitwise', 'poseidon'],
+    rcUnits: 4,
+    publicMemoryFraction: 8,
+    dilutedPool: {
+      unitsPerStep: 8,
+      spacing: 4,
+      nBits: 16,
+    },
+  },
+  all_cairo: {
+    builtins: [
+      'output',
+      'pedersen',
+      'range_check',
+      'ecdsa',
+      'bitwise',
+      'ec_op',
+      'keccak',
+      'poseidon',
+      'range_check96',
+    ],
+    rcUnits: 4,
+    publicMemoryFraction: 8,
+    dilutedPool: DEFAULT_DILUTED_POOL,
+  },
+  all_solidity: {
+    builtins: [
+      'output',
+      'pedersen',
+      'range_check',
+      'ecdsa',
+      'bitwise',
+      'ec_op',
+    ],
+    rcUnits: 8,
+    publicMemoryFraction: 8,
+    dilutedPool: DEFAULT_DILUTED_POOL,
+  },
+  dynamic: {
+    builtins: ['output'],
+    rcUnits: 16,
+    publicMemoryFraction: 8,
+    dilutedPool: DEFAULT_DILUTED_POOL,
+  },
+};
+
+export const ALL_LAYOUTS = Object.keys(layouts);
+
+export function isSubsequence(
+  subsequence: string[],
+  sequence: string[]
+): boolean {
+  return subsequence.reduce(
+    (acc, str) => {
+      const index = acc.sequence.findIndex((value) => value === str);
+      if (index === -1) {
+        acc.found = false;
+      } else {
+        acc.sequence = acc.sequence.slice(index + 1);
+      }
+      return acc;
+    },
+    { sequence, found: true }
+  ).found;
+}
```

Comment:
this looks a bit complicted to me. What about
so default empty sequence should return `true` as per your code. I propose

```
export function isSubsequence(
  subsequence: string[],
  sequence: string[]
): boolean {
    if (!sequence) return true;

    const index = sequence.find(subsequence[0])
    if (index === -1) return false;

    return sequence.slice(index, index + subsequence.length) === subsequence
}
```

Reply:
`!sequence` checks if sequence is undefined, to chech for empty array we have to check the value of `array.length`

It should be the length of `subsequence` rather than `sequence`, as the program builtins represent the subsequence while the sequence represent the layout builtins.

Array methods like `find` and `findIndex` takes a predicate as argument

Not sure to understand what does `slice(index, index + subsequence.length)` do, it should take all the remaining elements of `sequence` after the previously found `index`.
Plus, the `subsequence` might not exactly equal the reamining `sequence`.

But I got the idea, we can do it recursively, I'm proposing:

```typescript
export function isSubsequence(
  subsequence: string[],
  sequence: string[]
): boolean {
  if (!subsequence.length) return true;

  const index = sequence.findIndex((name) => name === subsequence[0]);
  if (index === -1) return false;

  return isSubsequence(subsequence.slice(1), sequence.slice(index + 1));
}
```